### PR TITLE
Update document version dropdown

### PR DESCRIPTION
**Pull request by smeng9**
**What's the problem this PR addresses?**

The versions displayed in the dropdown are outdated.
The link to some old versions are missing

**How did you fix it?**

Update version info and add links to old documentation site

**Checklist**

- [x] I have read the [Contributing Guide](https://yarnpkg.com/advanced/contributing).

- [x] I have set the packages that need to be released for my changes to be effective.

- [x] I will check that all automated PR checks pass before the PR gets reviewed.

### CODE REVIEW

File: packages/docusaurus/docusaurus.config.js [hunk 1]
```diff
@@ -229,11 +229,15 @@ const config = {
...
               {
-                label: `3.6.4`,
+                label: `3.8.1`,
                 href: `https://v3.yarnpkg.com`,
               },
               {
-                label: `1.22.19`,
+                label: `2.4.3`,
+                href: `https://v2.yarnpkg.com`,
+              },
```
Comment (clemyan) [inline packages/docusaurus/docusaurus.config.js:238]: There isn't a separate v2 docs. https://v2.yarnpkg.com is the v3 docs

Comment (smeng9) [reply packages/docusaurus/docusaurus.config.js:238]: Shall we still include the v2 versions?

Comment (clemyan) [reply packages/docusaurus/docusaurus.config.js:238]: I don't think so, given it is not an accurate documentation for v2

Comment (smeng9) [reply packages/docusaurus/docusaurus.config.js]: I have applied your suggestion for the documentation.

However, having https://v2.yarnpkg.com pointing to https://v3.yarnpkg.com is really confusing for the user. Shall we consider to host the documentation for v2 users somewhere while there are still some users using v2?

**Pull request closed without merging**